FileSpecs/Palettes.md

```diff
@@ -1,13 +1,37 @@
 # Palettes
 
-In the Amiga version there are 10 external palettes stored in COL_PALL.AMB. But there are other palettes as well. For example the [tilesets](IconData.md) store their own palette internally.
+Amberstar on the Amiga uses (at least) two palette representations:
+- wide palettes, which include a header and represent each color by 4 bytes
+- compact palettes, which lack a header and represent each color by 2 bytes
 
-The Amiga palettes have a total size of 66 (0x42) bytes. They start with a word which seems to give the number of colors. In Amberstar I've only seen 00 10 (= 16 colors) so far.
+In the Amiga version there are several palettes stored:
+- 10 wide palettes in `COL_PALL.AMB`, for [labyrinth data](LabData.md)
+- 26 compact palettes in `PICS80.AMB`, each following a raw pixmap
+- 2 wide palettes stored internally in each [tilesets](IconData.md)
+
+There may be other palettes stored elsewhere..
+
+## Wide Palettes
+
+Wide palettes have a total size of 66 (0x42) bytes. They start with a word which seems to give the number of colors. In Amberstar I've only seen 00 10 (= 16 colors) so far.
 
 After that the colors themselves follow. Each color is represented by 4 bytes. The first byte gives alpha, then red, then green and last blue. But alpha seems to be unused and always has a value of 0. The other color components are given by very low values in the range 0 to 7. You have to mulitply them by 32 to get the right color value. Maybe even add 16 in the end as well.
 
 Another approach would be `(value | (value << 4)) << 2`.
 
+## Compact palettes
+
+Compact palettes have a size of 32 (0x20) bytes.  They lack a header and store alpha, red, green, and blue information in one _nibble_ (half-byte) each.
+- alpha is `byte[0] & 0x70 >> 4`
```

Comment:
Are you sure that alpha is used? Moreover are you sure about the 7? I think I extracted some images already using F as a mask (4 bits per color component). Of course you have to do a `(value << 4) | value` to get the exact color. So 0x0123 would be R = 0x11, G = 0x22 and B = 0x33.

Reply:
Good point-- I haven't seen alpha used; that's a hold-over from an older version.  Should be removed.

The `... & 0x7` is something I noticed only with `PICS80.AMB`, where four pictures were off until I made this change.  I checked one of them against a YouTube video with the same image up, and it looked about right-- I couldn't see anything that was different with the additional bit set.

Reply:
Well there might be colors which just don't use the upper bit. Basically darker colors. I see no point in using lesser bits per pixel as the Amiga supports 4 bits per color component. But maybe there is some special trick involved. Who knows.

Reply:
The picture below illustrates the issue.  The images shown here are the four (of 26) images that use the MSB (i.e., `0x8`) of their R/G/B channel intensity nibbles in their associated palettes.
- The top row shows all pixels that use a colour with the MSB set in at least one channel (R/G/B).
- The middle row shows the result of interpreting the MSB as the MSB of the colour itself.  (Note that there is no arithmetic overflow happening here.)
- The bottom row shows the result of ignoring the MSB in those cases (not adjusted for brightness).

![image](https://user-images.githubusercontent.com/5527947/213918896-9eca9cd9-d79f-4732-8438-a1c4f6221758.png)

The MSB _might_ trigger some sort of special effect (perhaps flickering by updating the palette), but I am fairly sure that the bottom row looks more like the expected outcome than the middle row.

Reply:
You are right. I even documented it in a similar way before. But I only encountered images where the msb was never set.

Maybe it is some half bright indicator? I think you get the real color by multiplying (value & 0x7) by 32 (or left shift by 5). In general this results in values which are a bit shifted towards 0 than to 255 as the max value would be 7*32 which is 224 and not 255. So I guess adding 16 or even 31 might be desirable sometimes. Maybe the msb controls if adding those value should be performed or not. So 1011 would be 011b * 32 = 3 * 32 = 96. But as the msb is set, 31 (or 16?) is added to get 127 (or 112). In contrast the value 0011 maybe would just leave the value as 96?

Maybe you can check if this could match the colors.

Reply:
Another possibility would be that msb=0 means full bright (multiply by 32) and msb=1 means half bright (multiply only by 16).

Seems to be odd to do this but maybe it's beneficial when base colors should be matched somehow. Just a shot into the blue though.

Reply:
I experimented a bit with UAE but couldn't get it to do anything interesting with the extra bits-- it maps the Amiga palette registers to the palette entries * 2, and screenshots don't show any additional effects, either (i.e., UAE seems to interpret palette value `n` as R/G/B intensity `n * 0x22`).
Will include this in my next revision.